bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/AbstractExpression.java

```diff
@@ -356,13 +359,13 @@ protected void clearCandidates() {
         this.candidates = null;
     }
 
-    public ArrayList<E> getExpressionParts() {
+    public List<E> getExpressionParts() {
         return expressionParts;
```

Comment:
Should we return a immutable copy here so we detect other places where the internal expression list is changed without sync.?